iotedgedev/azurecli.py

```diff
@@ -157,24 +157,35 @@ def get_default_subscription(self):
 
     def get_subscription_id_starts_with(self, token):
         with output_io_cls() as io:
-            result = self.invoke_az_cli_outproc(["account", "list", "--query", "[?starts_with(@.id,'{0}') || starts_with(@.name,'{0}')] | [0]".format(token)],
+            result = self.invoke_az_cli_outproc(["account", "list", "--query", "[?starts_with(@.id,'{0}') || starts_with(@.name,'{1}')] | [0]".format(token.lower(), token)],
                                                 "Could not find a subscription that starts with '{0}'".format(token), io)
+
             if result:
                 out_string = io.getvalue()
                 if out_string:
+
                     data = json.loads(out_string)
-                    return data["id"]
+                    if len(data) == 1:
+                        return data[0]["id"]
+                    elif len(data) > 1:
+                        self.output.error("Found multiple subscriptions that start with '{0}'".format(token))
```

Comment:
Found multiple subscriptions that start with x. Please enter more characters to further refine your selection.